app/src/main/java/com/example/tasky/authentication/domain/FormValidator.kt

```diff
@@ -0,0 +1,9 @@
+package com.example.tasky.authentication.domain
+
+import javax.inject.Inject
+
+data class FormValidator @Inject constructor(
```

Comment:
I´d name it SignupFormValidator, or SignupUseCases

Since the name validator is specific to Signup

Also, this is a personal recommendation
But I prefer having
LoginUseCases /LoginFormValidator
and 
SignupUseCases / SignupFormValidator

Mostly to separate which ones goes where

Beucase if you use FormValidator for both, then Login is getting access to NameValidator, which tbh it shouldn´t really get access to

Reply:
I could rename it to be more specific but since Im only using FormValidator in SignUpViewModel I believe it doesnt make noise since LoginViewModel depends directly of EmailValidator

Reply:
Why only EmailValidator?
Wouldn't you want to also check if the password is valid?